velox/exec/DistinctAggregations.cpp

```diff
@@ -126,10 +143,37 @@ class TypedDistinctAggregations : public DistinctAggregations {
   }
 
  private:
+  TypePtr createInputType(const RowTypePtr rowType) const {
+    if (!isMultichannelDistinct_) {
+      return rowType->childAt(inputs_[0]);
+    }
+
+    // Otherwise, synthesize a ROW(distinct_channels[0..N])
+    std::vector<TypePtr> types;
+    std::vector<std::string> names;
+    for (column_index_t channelIndex : inputs_) {
+      names.emplace_back(rowType->names()[channelIndex]);
+      types.emplace_back(rowType->children()[channelIndex]);
+    }
+    return ROW(std::move(names), std::move(types));
+  }
+
+  [[nodiscard]] VectorPtr convertMultichannelInputToRow(
```

Comment:
The use of "multichannel" term is not ideal. How about renaming this method to makeInputForAccumulator and have it support both single-key and multi-key cases.